### ISSUE TITLE
Property 'observed' may not exist on type 'EventEmitter<any>'.

### ISSUE THREAD

**Issue by nickshoe** [KillerCodeMonkey/ngx-quill#1695]
This library used to work just fine in my Angular 13 app.
I've updated to Angular 14 (and Bootstrap 5.x) and now the event listeners doesn't work any longer:

```html
<quill-editor name="text"
              [placeholder]="placeholder"
              [modules]="editorModules"
              [styles]="{ height: '210px' }"
              (onBlur)="onBlur($event)"
              (onContentChanged)="onMessageTextChanged($event)">
...
</quill-editor>
```

```typescript
export class TextEditorComponent implements ControlValueAccessor {
  // ...

  onMessageTextChanged(contentChangedEvent) {
    this.textChanged.emit(contentChangedEvent.html);
    this.propagateChange(contentChangedEvent.html);
  }

  // ...
}
```

When I input text in the editor, the `textChangeHandler` method of the `QuillEditorBase` class gets invoked, but then this guard prevents the code from calling the event listener of my application:

This is the transpiled javascript version of this library:
```javascript
        this.textChangeHandler = (delta, oldDelta, source) => {
            // ...

            // only emit changes when there's any listener
            if (!this.onContentChanged.observed && !shouldTriggerOnModelChange) {
                return;
            }

            // ...
        }
```

https://github.com/KillerCodeMonkey/ngx-quill/blob/cea2e5bd4b70c1136889c47b0147d8c03a5b2874/projects/ngx-quill/src/lib/quill-editor.component.ts#L410

Using the debugger, I can actually see that the listener is indeed present, but since the `observed` property of the `EventEmitter` object seems to be non-existing, when accessed, it returns `undefined` and the guard makes the method return immediately.

![image](https://github.com/KillerCodeMonkey/ngx-quill/assets/32668766/dcf6cce9-96f3-447c-b6ff-c723e0613304)

As a fix, I suggest to use the `observers` property of the `EventEmitter` object and to check its length for presence of listeners:

```javascript
            // only emit changes when there's any listener
            if (this.onContentChanged.observers.length === 0 && !shouldTriggerOnModelChange) {
                return;
            }
```

**Comment by nickshoe** [comment]
Ok, I've found out what's the problem with my App: the `rxjs` version I'm using in my `package.json` is pretty old `^6.6.0`.

In the 6.x version of `rxjs` the `observed` property doesn't exist in the `Subject` class (which is defined in the `rxjs` package).

In the 7.x version of `rxjs`, used by Angular 15, the property indeed exists:
[```typescript
export class Subject<T> extends Observable<T> implements SubscriptionLike {
  // ...

  get observed() {
    return this.observers?.length > 0;
  }

  // ...
}](https://github.com/ReactiveX/rxjs/blob/456fc717e8910d4fbfcc9d3182c4a483703a1621/src/internal/Subject.ts#L105-L107)
```

**Comment by KillerCodeMonkey** [comment]
you are using the correct ngx-quill version for angular 14?

**Comment by nickshoe** [comment]
@KillerCodeMonkey I thought I did, since npm wasn't complaining about my old `rxjs` version during the upgrade process from Agular 13 to 14. I was pretty sure that I was using the 19.x version of `ngx-quill` module.. I guess I got confused somewhere in the middle of the update process.

I've just generated a brand new Angular 14 app with only `ngx-quill` v19.x (and its peer dependencies) and everything actually works just fine.

Hence, I'm closing the issue since it was due to a wrong peer dependency version (`rxjs` too old) anyways. \
Thanks for your time!
